packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts

```diff
@@ -157,6 +160,7 @@ export default createRule<Options, MessageIds>({
     function prettierFormat(
       code: string,
       location: TSESTree.Node,
+      silent?: boolean,
```

Comment:
[Refactor] Hmm, I'm not generally a fan of adding boolean parameters to modify core behavior of a function. It makes it hard to keep track of the function's responsibilities & adds to its complexity. I think the code would be easier to wrangle if we extract out the logic we actually care about from the existing function into a new one.

How about this  of function signatures?

* `function getCodeFormatted(code: string): string | FormattingError | null`
* `function getCodeFormattedOrReport(code: string, location: TSESTree.Node): string | null`

...where `FormattingError` indicates the error message, `getCodeFormatted` contains the call to Prettier, and `getCodeFormattedOrReport` contains the call to `context.report`?